src/charm.py

```diff
@@ -142,6 +164,53 @@ def _run_sql_migration(self, set_timeout: bool) -> None:
         stdout, _ = process.wait_output()
         logger.info(f"Executing automigration: {stdout}")
 
+    def _update_config_restart_service(self, event: HookEvent) -> None:
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Hydra container. Deferring the event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Hydra container")
+            return
+
+        self.unit.status = MaintenanceStatus("Configuring resources")
+
+        try:
```

Comment:
This try/except block can be in its own `_hydra_service_is_running` property/method:

```python
if not self._hydra_service_is_running:
    event.defer()
    return
```